pennylane/devices/preprocess.py

```diff
@@ -48,9 +48,12 @@ def _operator_decomposition_gen(
     max_expansion: Optional[int] = None,
     current_depth=0,
     name: str = "device",
-    error: Exception = DeviceError,
+    error: Optional[Exception] = None,
```

Comment:
Was there a need for this? Couldn't we leave it as DeviceError since is not mutable.

Reply:
Just a safety measure against the pytest namespace issue that we discussed before

Reply:
Discussed in person: this change is not necessary but also not blocking. Closing this comment.